spyder/app/mainwindow.py

```diff
@@ -1660,84 +1658,86 @@ def open_external_file(self, fname):
-            self.not_active_path = tuple(name for name in not_active_path
-                                         if osp.isdir(name))
-
-    def save_python_path(self, new_path_dict):
-        """
-        Save path in Spyder configuration folder.
-
-        `new_path_dict` is an OrderedDict that has the new paths as keys and
-        the state as values. The state is `True` for active and `False` for
-        inactive.
         """
-        path = [p for p in new_path_dict]
-        not_active_path = [p for p in new_path_dict if not new_path_dict[p]]
-        try:
-            encoding.writelines(path, self.SPYDER_PATH)
-            encoding.writelines(not_active_path, self.SPYDER_NOT_ACTIVE_PATH)
-        except EnvironmentError as e:
-            logger.error(str(e))
-        CONF.set('main', 'spyder_pythonpath', self.get_spyder_pythonpath())
+        Load paths stored in Spyder configuration folder, add them to the
+        modern configuration, and remove the obsolete files.
 
-    def get_spyder_pythonpath_dict(self):
+        TODO: Remove on later release
         """
-        Return Spyder PYTHONPATH.
+        SPYDER_PATH = get_conf_path('path')
+        SPYDER_NOT_ACTIVE_PATH = get_conf_path('not_active_path')
 
-        The returned ordered dictionary has the paths as keys and the state
-        as values. The state is `True` for active and `False` for inactive.
+        path_dict = OrderedDict()
+        if osp.isfile(SPYDER_PATH):
+            with open(SPYDER_PATH, 'r', encoding='utf-8') as f:
+                paths = f.read().splitlines()
+            for path in paths:
+                if osp.isdir(path):
+                    path_dict[path] = True
+            os.remove(SPYDER_PATH)  # No longer use file
+            logger.info(f"Removed obsolete '{SPYDER_PATH}'")
+
+        if osp.isfile(SPYDER_NOT_ACTIVE_PATH):
+            with open(SPYDER_NOT_ACTIVE_PATH, 'r', encoding='utf-8') as f:
+                paths = f.read().splitlines()
+            for path in paths:
+                if osp.isdir(path):
+                    path_dict[path] = False
+            os.remove(SPYDER_NOT_ACTIVE_PATH)  # No longer use file
+            logger.info(f"Removed obsolete '{SPYDER_NOT_ACTIVE_PATH}'")
+
+        if path_dict:
+            self.update_python_path(path_dict)
 
-        Example:
-            OrderedDict([('/some/path, True), ('/some/other/path, False)])
+    def get_spyder_pythonpath(self):
         """
-        self.load_python_path()
-
-        path_dict = OrderedDict()
-        for path in self.path:
-            path_dict[path] = path not in self.not_active_path
+        Return Spyder PYTHONPATH, including project paths, as ordered
+        dictionary.
 
-        for path in self.project_path:
+        The returned dictionary has the paths as keys and the state as values.
+        The state is `True` for active and `False` for inactive.
+        """
+        path_dict = OrderedDict(CONF.get('pythonpath_manager', 'paths'))
```

Comment:
This shows to me that what you want is a new plugin because you're adding a new section in our config system to handle this: `pythonpath_manager`.